src/main/kotlin/com/team4099/robot2022/subsystems/Shooter.kt

```diff
@@ -0,0 +1,75 @@
+package com.team4099.robot2022.subsystems
+
+import com.ctre.phoenix.motorcontrol.ControlMode
+import com.ctre.phoenix.motorcontrol.can.TalonFX
+import com.team4099.lib.logging.Logger
+import com.team4099.lib.units.ctreAngularMechanismSensor
+import com.team4099.robot2022.config.Constants
+import com.team4099.robot2022.config.Constants.ShooterConstants
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+
+object Shooter : SubsystemBase() {
+  private val leaderMotor = TalonFX(ShooterConstants.LEADER_MOTOR_ID)
+  private val followerMotor = TalonFX(ShooterConstants.FOLLOWER_MOTOR_ID)
+
+  private val shooterSensor = ctreAngularMechanismSensor(leaderMotor, 2048, 1.0)
+
+  var shooterState = ShooterConstants.ShooterState.IDLE
+    set (value) {
+      when (value) {
+        ShooterConstants.ShooterState.IDLE -> {
+          leaderMotor.set(ControlMode.PercentOutput, 0.0)
+        }
+        ShooterConstants.ShooterState.SPIN_UP -> {
+          leaderMotor.set(ControlMode.PercentOutput, 1.0)
```

Comment:
we shouldn't have sutff like 0.0 in Shooter.kt. there is a value already for each of the enums so you may as well instead do IDLE.value

Reply:
oh wait we shouldnt even be using the .set method for shooterState as we want to launch the ball at a certain velocity so that means we use PID